parcels/particleset.py

```diff
@@ -278,6 +279,37 @@ def from_field(cls, fieldset, pclass, start_field, size, mode='monte_carlo', dep
 
         return cls(fieldset=fieldset, pclass=pclass, lon=lon, lat=lat, depth=depth, time=time, lonlatdepth_dtype=lonlatdepth_dtype, repeatdt=repeatdt)
 
+    @classmethod
+    def from_particlefile(cls, fieldset, pclass, filename, repeatdt=None, lonlatdepth_dtype=None):
+        """Initialise the ParticleSet from a ParticleFile
+
+        :param fieldset: :mod:`parcels.fieldset.FieldSet` object from which to sample velocity
+        :param pclass: mod:`parcels.particle.JITParticle` or :mod:`parcels.particle.ScipyParticle`
```

Comment:
I think we need a bit more explanation in the docstring. What does the method exactly do?
Like, it looks in the code you take the last time step of the data, so:
* if all particles are started at t=0 and some are deleted at t=3 and other at t=4, we only keep the one at t=4, right?
but if particles are started at t=[0,1,2,3] and are all deleted after 5 days of runtime, you will load all particles.
* Should we only keep particles with `time=max(pfile.variables['time']` (or min if `pfile.variables['time'[0,1] < pfile.variables['time'[0,0]`) ?
* Should we mention that particles not started yet won't be exported and then won't be reloaded, such that `from_particlefile` is not the exact restart of the simulation as it was before quitting (I don't think it should, but important to say it)
* Mentioning that p.id are updated and not the one of the original particles
* Should we add a `time` param such that we load the particles at some time of the simulation? That might be cool actually (with default=None, when we load the last time step)

Reply:
All very good points! You thought this through much more than I did. I will keep working on this implementation then